tests/conftest.py

```diff
@@ -120,3 +122,36 @@ def enable_feast_integration_fixture():
     """
     with patch.dict(os.environ, {"FEATUREBYTE_FEAST_INTEGRATION_ENABLED": "True"}):
         yield
+
+
+def pytest_xdist_make_scheduler(config, log):
+    """Group tests from the same module together"""
+
+    class GroupByModuleScheduler(LoadScopeScheduling):
+        """Group tests from the same module together and sort the test cases by module & test names"""
+
+        def _assign_work_unit(self, node):
+            """Assign a work unit to a node."""
+            assert self.workqueue
+
+            # Grab a unit of work
+            scope, work_unit = self.workqueue.popitem(last=False)
+
+            # Change the default from OrderedDict() to SortedDict()
+            assigned_to_node = self.assigned_work.setdefault(node, default=SortedDict())
+            assigned_to_node[scope] = work_unit
+
+            # Ask the node to execute the workload
+            worker_collection = self.registered_collections[node]
+            nodeids_indexes = [
+                worker_collection.index(nodeid)
+                for nodeid, completed in work_unit.items()
+                if not completed
+            ]
+
+            node.send_runtest_some(nodeids_indexes)
+
+        def mark_test_pending(self, item):
+            raise NotImplementedError()
+
+    return GroupByModuleScheduler(config, log)
```

Comment:
main branch:
```
(featurebyte-py3.8) ➜  featurebyte git:(main) ✗ poetry run pytest -vv tests/unit/service/test_feature_materialize.py tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments -n2
================================================================================================================================================ test session starts ================================================================================================================================================
platform darwin -- Python 3.8.18, pytest-7.4.3, pluggy-1.3.0 -- /Users/senbonggee/Library/Caches/pypoetry/virtualenvs/featurebyte-Ryj2vg6L-py3.8/bin/python
cachedir: .pytest_cache
rootdir: /Users/senbonggee/Workspace/featurebyte
configfile: pyproject.toml
plugins: timeout-2.2.0, asyncio-0.19.0, cov-4.1.0, rerunfailures-11.1.2, typeguard-2.13.3, xdist-3.5.0, anyio-4.2.0
asyncio: mode=strict
2 workers [11 items]    
scheduling tests via LoadScheduling

tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[False] 
tests/unit/service/test_feature_materialize.py::test_materialize_features 
[gw0] [  9%] PASSED tests/unit/service/test_feature_materialize.py::test_materialize_features 
[gw1] [ 18%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[False] 
tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[True] 
tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features_if_materialized_before 
[gw0] [ 27%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[True] 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[True] 
[gw1] [ 36%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features_if_materialized_before 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[False] 
[gw1] [ 45%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[False] 
[gw0] [ 54%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[True] 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__databricks_unity[False] 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_exists 
[gw1] [ 63%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__databricks_unity[False] 
tests/unit/service/test_feature_materialize.py::test_drop_columns 
[gw0] [ 72%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_exists 
tests/unit/service/test_feature_materialize.py::test_drop_table 
[gw1] [ 81%] PASSED tests/unit/service/test_feature_materialize.py::test_drop_columns 
tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments 
[gw0] [ 90%] PASSED tests/unit/service/test_feature_materialize.py::test_drop_table 
[gw1] [100%] PASSED tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments 
```
current branch:
```
(featurebyte-py3.8) ➜  featurebyte git:(senbong/fix-test-coupling) ✗ poetry run pytest -vv tests/unit/service/test_feature_materialize.py tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments -n2
================================================================================================================================================ test session starts ================================================================================================================================================
platform darwin -- Python 3.8.18, pytest-7.4.3, pluggy-1.3.0 -- /Users/senbonggee/Library/Caches/pypoetry/virtualenvs/featurebyte-Ryj2vg6L-py3.8/bin/python
cachedir: .pytest_cache
rootdir: /Users/senbonggee/Workspace/featurebyte
configfile: pyproject.toml
plugins: timeout-2.2.0, asyncio-0.19.0, cov-4.1.0, rerunfailures-11.1.2, typeguard-2.13.3, xdist-3.5.0, anyio-4.2.0
asyncio: mode=strict
2 workers [11 items]    
scheduling tests via GroupByModuleScheduler

tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments 
tests/unit/service/test_feature_materialize.py::test_materialize_features 
[gw1] [  9%] PASSED tests/unit/service/test_offline_feature_store_comment.py::test_apply_comments 
[gw0] [ 18%] PASSED tests/unit/service/test_feature_materialize.py::test_materialize_features 
tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[True] 
[gw0] [ 27%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[True] 
tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[False] 
[gw0] [ 36%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features[False] 
tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features_if_materialized_before 
[gw0] [ 45%] PASSED tests/unit/service/test_feature_materialize.py::test_scheduled_materialize_features_if_materialized_before 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[True] 
[gw0] [ 54%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[True] 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[False] 
[gw0] [ 63%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_does_not_exist[False] 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_exists 
[gw0] [ 72%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__table_exists 
tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__databricks_unity[False] 
[gw0] [ 81%] PASSED tests/unit/service/test_feature_materialize.py::test_initialize_new_columns__databricks_unity[False] 
tests/unit/service/test_feature_materialize.py::test_drop_columns 
[gw0] [ 90%] PASSED tests/unit/service/test_feature_materialize.py::test_drop_columns 
tests/unit/service/test_feature_materialize.py::test_drop_table 
[gw0] [100%] PASSED tests/unit/service/test_feature_materialize.py::test_drop_table 
```

Reply:
After adding the scheduling, tests of same module is grouped into same worker (check `[gw0]` and `[gw1]`).